corehq/apps/geospatial/utils.py

```diff
@@ -31,16 +31,22 @@ def _format_coordinates(lat, lon):
     return f"{lat} {lon} 0.0 0.0"
 
 
-def set_case_gps_property(domain, case_data):
+def set_case_gps_property(domain, case_data, create_case=False):
     location_prop_name = get_geo_case_property(domain)
-    helper = CaseHelper(domain=domain, case_id=case_data['id'])
-    case_data = {
+    data = {
         'properties': {
             location_prop_name: _format_coordinates(case_data['lat'], case_data['lon'])
         }
     }
 
-    helper.update(case_data)
+    if create_case:
```

Comment:
It feels weird to create the case in this function. Maybe either
1. Update the function name to something like `upsert_case_with_gps_property` (preferred option) 
2. Create the case before calling this function

Reply:
I'm also opted to go for option 1, but we could also split this out into two different functions with one for update and one for create. @Charl1996 What are your thoughts?

Reply:
Splitting it could work

Reply:
[eb20b1c](https://github.com/dimagi/commcare-hq/pull/33673/commits/eb20b1cd655acd396d7bf44df817ad97b76d5fc1)